client/http4s-client/src/main/scala/sttp/tapir/client/http4s/EndpointToHttp4sClient.scala

```diff
@@ -147,6 +149,9 @@ private[http4s] class EndpointToHttp4sClient(clientOptions: Http4sClientOptions)
       case RawBodyType.InputStreamBody =>
         val entityEncoder = EntityEncoder.inputStreamEncoder[F, InputStream]
         req.withEntity(Applicative[F].pure(encoded.asInstanceOf[InputStream]))(entityEncoder)
```

Comment:
**question:** @adamw shouldn't this be defered with a `Sync[F].delay` instead of `Applicative[F].pure`?

Reply:
but `encoded` is a "pure" value, there's no computation happening here?

Reply:
Oh, yeah, right. It only applies to the next case, where resolving `.inputStream()` from the `encoded` can be delayed.